# Cytoscape: Network Biology, topological and pathway analysis, and data visualization  
### Grace Constantian, Michael Brown, and Christy Deliivanov

*   [Introduction](#introduction)
*   [Applications](#applications)
*   [Example](#example)
*   [Sources](#sources)

* * *

## Introduction

### What is Cytoscape?

Cytoscape was created in 2002 by a group of researchers including Paul Shannon, Andrew Markiel, Owen Ozier, Nitin S. Baliga, Jonathan T. Wang, Daniel Ramage, Nada Amin, Benno Schwikowski, and Trey Ideker as a tool for network biology research. Now while it is still used as a vital tool for network biology research it has broader applications to include fields such as social science, systems biology, and computational chemistry. As of 2023, Cytoscape has grown in popularity with about 20,000 downloads per month. The Cytoscape website refers to itself as 'an open source software platform for visualizing molecular interaction networks and biological pathways and integrating these networks with annotations, gene expression profiles and other state data'(https://cytoscape.org/). 

Cytoscape's architecture includes a core and additional apps you can plug in for enhanced data integration, analysis, and visualization. Cytoscape core gives the user basic functionality to layout and query their data and to visually integrate the network with state data. For more complex projects plugins might be required to analyze and visualize data.

Cytoscape is a non-profit, open source project that is sponsored on public and private grants. Whenever using Cytoscape for a project it is important to cite the original paper. To access the paper click this link: (https://genome.cshlp.org/content/13/11/2498.full.pdf+html)

The current maintenance and development of Cytoscape is being funded by the U.S. National Human Genome Research Intitute (NHGRI).

### What are the different apps in Cytoscape?

Apps (formerly called Plugins) are additional features you can add to your Cytoscape core that can be found on the Cytoscape website. These apps can be created by anyone and the creators of Cytoscape encourage people to develop apps. 

To install an app follow the steps below:

1. Go to the Cytoscape App Store and find the app you want: https://apps.cytoscape.org/
2. Open Cytoscape on your computer and go to the menu bar, choose Apps -> App Manager
3. Select the Install Apps tab and in the Download Site box type in "http://apps.cytoscape.org/"
4. In the search box type in the name of the app you would like to download
5. Click the Install button to install the app

You can also uninstall apps or temporarily disable apps as well. To do this go to the 'Currently Installed' tab in the app manager to find the app and then select uninstall or disable depending on your need. 

The Top 6 Apps on Cytoscape are:

1. ClueGo - Creates and Visualizes the non-redundant terms for large clusters of genes in a functinoally grouped network. Performs single cluster analysis and comparison of multiple clusters.
2. StringApp - Imports functional associations or physical interactions between protein-chemical and protein-protein pairs from a variety of sources. Users can provide a list of genes, proteins, compounds, diseases, species etc and the app queries the database to return a matching network.
3. MCODE - Clusters a network based on topology to find highly interconnected regions.
4. yFiles Layout Algorithms - Connects layout algorithms from the yFiles diagramming libraries to your Cytoscape.
5. CluePedia - Search tool for new markers that are potentially associated to a variety of pathways. CluePedia calculates both linear and non-linear statistical dependencies from data. CluePedia depends on the ClueGo app, so make sure both are installed.
6. BiNGO - Calculates overrepresented Gene Ontology categories in a set of genes and displays them as a network of significant Gene Ontology terms.

### What is Network Biology?

* * *

## Applications


* * *

## Example

* * *

## Sources

*
