# Cytoscape: Network Biology, topological and pathway analysis, and data visualization  
### Grace Constantian, Michael Brown, and Christy Deliivanov

*   [Introduction](#introduction)
*   [Applications](#applications)
*   [Example](#example)
*   [Sources](#sources)

* * *

## Introduction

### What is Cytoscape?

Cytoscape was created in 2002 by a group of researchers including Paul Shannon, Andrew Markiel, Owen Ozier, Nitin S. Baliga, Jonathan T. Wang, Daniel Ramage, Nada Amin, Benno Schwikowski, and Trey Ideker as a tool for network biology research. Now while it is still used as a vital tool for network biology research it has broader applications to include fields such as social science, systems biology, and computational chemistry. As of 2023, Cytoscape has grown in popularity with about 20,000 downloads per month. The Cytoscape website refers to itself as 'an open source software platform for visualizing molecular interaction networks and biological pathways and integrating these networks with annotations, gene expression profiles and other state data'(https://cytoscape.org/). 

Cytoscape's architecture includes a core and additional apps you can plug in for enhanced data integration, analysis, and visualization. Cytoscape core gives the user basic functionality to layout and query their data and to visually integrate the network with state data. For more complex projects plugins might be required to analyze and visualize data.

Cytoscape is a non-profit, open source project that is sponsored on public and private grants. Whenever using Cytoscape for a project it is important to cite the original paper. To access the paper click this link: (https://genome.cshlp.org/content/13/11/2498.full.pdf+html)

The current maintenance and development of Cytoscape is being funded by the U.S. National Human Genome Research Intitute (NHGRI).

### What are the different apps in Cytoscape?

### What is Network Biology?

* * *

## Applications


* * *

## Example

* * *

## Sources

*
